src/main/java/sergey/EquationServlet.java

```diff
@@ -0,0 +1,194 @@
+                }
+            }
+            for (String key : StringValue.keySet()) {
+                Integer value = args.get(StringValue.get(key));
+                args.put(key, value);
+            }
+            int result = calculate(tokens, args);
+
+//            System.out.println("args: " + args.toString());
+//            System.out.println("result: " + result);
+
+            resp.setStatus(SC_OK);
+            resp.getOutputStream().print(result);
+            resp.getOutputStream().flush();
+            resp.getOutputStream().close();
+        } catch (ParseException e) {
+            resp.sendError(SC_CONFLICT, "Problem with equation");
+        } catch (NumberFormatException ee) {
+            resp.sendError(SC_CONFLICT, "Problem with variable");
+        } catch (NullPointerException eee) {
+            resp.sendError(SC_CONFLICT, "Not set required parameters");
+        }
+    }
+
+    private List<String> parseIntoPolishNotation(String equation) throws ParseException {
+        if (!(equation.contains("+")
+                || equation.contains("-")
+                || equation.contains("*")
+                || equation.contains("/"))) {
+            throw new ParseException("notValidEquation", 0);
+        }
+        List<String> tokens = new ArrayList<>();
+        StringTokenizer tokenizer = new StringTokenizer(equation.replaceAll(" ", ""), "()+-/*", true);
+        while (tokenizer.hasMoreElements()) {
+            tokens.add(tokenizer.nextToken());
+        }
+        Deque<String> stack = new ArrayDeque<>();
+        List<String> polishNotation = new ArrayList<>();
+        for (String token : tokens) {
+            try {
+                switch (token) {
+                    case "+":
+                    case "-":
+                    case ")":
+                        while ("+-/*".contains(stack.getFirst())) {
+                            polishNotation.add(stack.removeFirst());
+                        }
+                        if (token.equals(")")) {
+                            if (!stack.getFirst().equals("("))
+                                throw new ParseException("notValidEquation", 0);
+                            stack.removeFirst();
+                        } else {
+                            stack.addFirst(token);
+                        }
+                        break;
+                    case "*":
+                    case "/":
+                        while ("/*".contains(stack.getFirst())) {
+                            polishNotation.add(stack.removeFirst());
+                        }
+                        stack.addFirst(token);
+                        break;
+                    case "(":
+                        stack.addFirst(token);
+                        break;
+                    default:
+                        polishNotation.add(token);
+                }
+            } catch (NoSuchElementException e) {
+                if (!token.equals(")"))
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Position literals first in String comparisons](https://app.codacy.com/manual/thejerome/IFMO_Servlets_20190915/pullRequest?prid=4723796)